.github/workflows/build.yml

```diff
@@ -73,52 +73,18 @@ jobs:
                 ;;
             esac
           fi
-
-          echo ::set-output name=os_ref::$OS_REF
-          echo ::set-output name=apps_ref::$APPS_REF
-          echo ::set-output name=testing_ref::$TESTING_REF
-
-      - name: Checkout nuttx repo
-        uses: actions/checkout@v2
-        with:
-          repository: apache/incubator-nuttx
-          ref: ${{ steps.gittargets.outputs.os_ref }}
-          path: sources/nuttx
-          fetch-depth: 1
-
-      - name: Checkout apps repo
-        uses: actions/checkout@v2
+          echo "#!/bin/sh" > gitenv.sh
+          echo "echo ::set-env name=os_ref::$OS_REF" >> gitenv.sh
+          echo "echo ::set-env name=apps_ref::$APPS_REF" >> gitenv.sh
+          echo "echo ::set-env name=testing_ref::$TESTING_REF" >> gitenv.sh
+      - name: Upload gitrefs
+        uses: actions/upload-artifact@v2
         with:
-          repository: apache/incubator-nuttx-apps
-          ref: ${{ steps.gittargets.outputs.apps_ref }}
-          path: sources/apps
-          fetch-depth: 1
-
-      - name: Checkout testing repo
-        uses: actions/checkout@v2
-        with:
-          repository: apache/incubator-nuttx-testing
-          ref: ${{ steps.gittargets.outputs.testing_ref }}
-          path: sources/testing
-          fetch-depth: 1
-
-      - name: Create Source Bundle
-        run: tar -czf sources.tar.gz sources
-      - name: Archive Source Bundle
-        uses: actions/upload-artifact@v1
-        with:
-          name: source-bundle
-          path: sources.tar.gz
-
-      - name: Cache Source
-        id: cache-source
-        uses: actions/cache@v1
-        with:
-          path: sources
-          key: build-sources-${{ github.run_id }}
+          name: gitrefs
+          path: gitenv.sh
 
   Linux:
-    needs: Fetch-Source
+    needs: Compute-Source-Refs
```

Comment:
But Compute-Source-Refs don't generate and upload the cache anymore, why don't we move os_ref/apps_ref/testing_ref into a common function? So we can invoke it in macOS/Linux directly. The current workflow which upload/download gitenv.sh seems too complex.

Reply:
This is how github recommends that you pass information between jobs. Once they implement templating like every other yaml based CI system we can remove this.

As for why it is not a function. At this point prior to doing cloning there is not repo to put the script in, only the workflow yaml so it must be in the yaml. Making it a job that others depend on saves us from having to copy it to every job (currently Linux and macOS, but soon windows as well)

As for the bit about the cache, I had the cache before because it was much faster to move the larger git repos. Now we just move 100 bytes of data so there is no reason to do the cache since we had to support the fallback of artifacts anyway.

Reply:
> This is how github recommends that you pass information between jobs. Once they implement templating like every other yaml based CI system we can remove this.
> 
> As for why it is not a function. At this point prior to doing cloning there is not repo to put the script in, only the workflow yaml so it must be in the yaml. Making it a job that others depend on saves us from having to copy it to every job (currently Linux and macOS, but soon windows as well)
> 

I mean the funciton in action yml not in bash script. Maybe the step is better term than function in this context.

> As for the bit about the cache, I had the cache before because it was much faster to move the larger git repos. Now we just move 100 bytes of data so there is no reason to do the cache since we had to support the fallback of artifacts anyway.

sharing the code by job need save the result to script and then load/execute this script in other jobs. Can we use the common step to share the code like:
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions
```
Example using action in the same repository as the workflow
./path/to/dir

The path to the directory that contains the action in your workflow's repository. You must check out your repository before using the action.

jobs:
  my_first_job:
    steps:
      - name: Check out repository
        uses: actions/checkout@v2
      - name: Use local my-action
        uses: ./.github/actions/my-action
```
Or even put the common step in the same yml file:
https://github.com/actions/checkout/blob/master/action.yml
```
name: 'Checkout'
description: 'Checkout a Git repository at a particular version'
...
runs:
  using: node12
  main: dist/index.js
  post: dist/index.js
```
I guess that Compute-Source-Refs become a common step if we move Compute-Source-Refs out of the job.
From my understand, once the common code become to a step, we can use it inside Linux/macOS job and get the result directly.

Reply:
That is a good option in the since that it actually can be stored in the testing repo (see what I did for the Linux build step). But it means that we would need to write it in nodejs since you cannot use the docker action on macOS or Windows. I was trying to avoid writing nodejs from a maintenance standpoint. I can try it if you think it's ok.

Reply:
Can we use bash script here?

Reply:
Nope only if you make it a Docker action which limits you to Linux. Our only option would be JavaScript.
https://help.github.com/en/actions/building-actions/about-actions#types-of-actions